### PR TITLE
fix space visibility by fetching correct room for join rules

### LINKED ISSUE #2085: Limited space visibility : Space with access selection not working correctly
https://github.com/user-attachments/assets/abbf43e1-6b0b-4fc9-a48c-01a1071e8d69

### PR DESCRIPTION
**Pull request by gtalha07**
fixes #2085.

 
**Could probably fix [meta#523](https://github.com/acterglobal/a3-meta/issues/523) because not reproducible here. Needs Confirmation.**

Preview:

https://github.com/user-attachments/assets/143257ac-8090-4fd1-a190-5e93342b92a9

### CODE REVIEW

File: app/lib/features/space/settings/pages/visibility_accessibility_page.dart [hunk 1]
```diff
@@ -249,7 +249,7 @@ class _VisibilityAccessibilityPageState
       final spaceId = await selectSpaceDrawer(context: context);
       if (spaceId != null) {
         final spaceList =
-            await ref.read(joinRulesAllowedRoomsProvider(spaceId).future);
+            await ref.read(joinRulesAllowedRoomsProvider(roomId).future);
         final isAlreadyAdded = spaceList.any((roomId) => roomId == spaceId);
```
Comment (kumarpalsinh25) [inline app/lib/features/space/settings/pages/visibility_accessibility_page.dart:253]: What is different between `roomId` and `widget.roomId`?

**Review by kumarpalsinh25 — APPROVED** [review]
Nothing much to review from code-wise. Added minor comment to address before merge.

File: app/lib/features/space/settings/pages/visibility_accessibility_page.dart (hunk 1 above)
Comment (gtalha07) [reply app/lib/features/space/settings/pages/visibility_accessibility_page.dart:253]: the ```roomId``` is the id we selected from drawer while ```widget.roomId``` is the room identifier we've opened settings.

Comment (kumarpalsinh25) [reply app/lib/features/space/settings/pages/visibility_accessibility_page.dart:253]: When I look at entire code file, it looks like both are same. widget.roomId pass to the function argument where roomId is the param.

**Pull request merged**